src/shogun/util/factory.h

```diff
@@ -99,6 +102,7 @@ namespace shogun
 	BASE_CLASS_FACTORY(EvaluationResult, evaluation_result)
 	BASE_CLASS_FACTORY(Distribution, distribution)
 	BASE_CLASS_FACTORY(CombinationRule, combination_rule)
+	BASE_CLASS_FACTORY(GaussianProcessMachine, gaussian_process)
```

Comment:
Should it be its own class? Does it not work using Machine? I guess you need a method to compute the marginal likelihood?

Reply:
gp need `get_variance_vector`, 'get_probabilities'..., i guess it's not a good idea that add those methods to the Machine?

Reply:
Not to machine, no. I guess it makes sense to have GPs as it's own thing, like we have SVMs.. @karlnapf?

Reply:
Yes. But let's call it "GaussianProcess"

Reply:
@LiuYuHui could you also do the renaming suggested by @karlnapf please?

Reply:
or is `gaussian_process` fine @karlnapf ?

Reply:
for the factory, yes. But then let's also rename the base class so that c++ and swig will look similar
`gaussian_process(...)` and `make_shared<GaussianProcess>(...)`

Reply:
after #4995 is merged, i will rename `gaussian_process` to `create_gaussian_process`, and use create<GaussianProcess> in C++ side.